src/active_set_quadratic_direct_solve_wolfe.jl

```diff
@@ -0,0 +1,401 @@
+
+function active_set_argmin(as::ActiveSetQuadraticLinearSolveWolfe, direction)
+    return active_set_argmin(as.active_set, direction)
+end
+
+function active_set_argminmax(as::ActiveSetQuadraticLinearSolveWolfe, direction; Φ=0.5)
+    return active_set_argminmax(as.active_set, direction; Φ=Φ)
+end
+
+# generic quadratic with quadratic information provided
+
+"""
+    solve_quadratic_activeset_lp!(as::ActiveSetQuadraticLinearSolveWolfe{AT, R, IT, H}))
+
+Solves the auxiliary LP over the current active set.
+The method is specialized by type `H` of the Hessian matrix `A`.
+"""
+function solve_quadratic_activeset_lp!(
+    as::ActiveSetQuadraticLinearSolveWolfe{AT,R,IT,<:AbstractMatrix},
+) where {AT,R,IT}
+    nv = length(as)
+    o = as.lp_optimizer
+    MOI.empty!(o)
+    λ = MOI.add_variables(o, nv)
+    # λ ≥ 0, ∑ λ == 1
+    # MOI.add_constraint.(o, λ, MOI.GreaterThan(0.0))
+    sum_of_variables = MOI.ScalarAffineFunction(MOI.ScalarAffineTerm.(1.0, λ), 0.0)
+    MOI.add_constraint(o, sum_of_variables, MOI.EqualTo(1.0))
+    # Vᵗ A V λ == -Vᵗ b
+    for atom in as.atoms
+        lhs = MOI.ScalarAffineFunction{Float64}([], 0.0)
+        Base.sizehint!(lhs.terms, nv)
+        # replaces direct sum because of MOI and MutableArithmetic slow sums
+        for j in 1:nv
+            push!(lhs.terms, MOI.ScalarAffineTerm(fast_dot(atom, as.A, as.atoms[j]), λ[j]))
+        end
+        rhs = -dot(atom, as.b)
+        MOI.add_constraint(o, lhs, MOI.EqualTo(rhs))
+    end
+    MOI.set(o, MOI.ObjectiveFunction{typeof(sum_of_variables)}(), sum_of_variables)
+    MOI.set(o, MOI.ObjectiveSense(), MOI.MIN_SENSE)
+    MOI.optimize!(o)
+    if MOI.get(o, MOI.TerminationStatus()) ∉ (MOI.OPTIMAL, MOI.FEASIBLE_POINT, MOI.ALMOST_OPTIMAL)
+        return as
+    end
+
+    # Wolfe's pull back
+    ## Extract lambda values into a vector
+    lambda_values = Vector{Float64}(undef, length(λ))
+    for idx in eachindex(λ)
+        lambda_values[idx] = MOI.get(o, MOI.VariablePrimal(), λ[idx])
+    end
+
+    ## Load old weights into vector and do ratio test
+    old_weights = as.weights
+    tau_min = 1.0
+    for i in 1:length(lambda_values)
+        if lambda_values[i] < old_weights[i]
+            tau_min = min(tau_min, old_weights[i] / (old_weights[i] - lambda_values[i]))
+        end
+    end
+    tau = tau_min
+
+    # println("lambda_values: ", lambda_values)
+    # println("tau: ", tau)
+
+    ## Compute new lambdas
+    new_lambdas = Vector{R}(undef, length(lambda_values))
+    for i in 1:length(lambda_values)
+        new_lambdas[i] = (1 - tau) * old_weights[i] + tau * lambda_values[i]
+    end
+
+    @assert all(0 .<= new_lambdas .<= 1) "All new_lambdas must be between 0 and 1"
+    @assert isapprox(sum(new_lambdas), 1.0, atol=1e-5) "The sum of new_lambdas must be approximately 1"
+
+    indices_to_remove = Int[]
+    new_weights = R[]
+    for idx in eachindex(λ)
+        weight_value =  new_lambdas[idx] # using new lambdas
```

Comment:
this should implement wolfe's pullback. get the unconstrained solution. form the line segment between current solution and unconstrained solution. compute tau on the line so that you stay feasible. update lambda and use